nx-X11/programs/Xserver/hw/nxagent/Window.c

```diff
@@ -3111,6 +3135,18 @@ FIXME: Do we need to set save unders attribute here?
     }
   }
 
+  if ((nxagentRealWindowProp) && (!nxagentWindowTopLevel(pWin)))
+  {
+    Atom prop = MakeAtom("NX_REAL_WINDOW", strlen("NX_REAL_WINDOW"), True);
+
+    if (ChangeWindowProperty(pWin, prop, XA_WINDOW, 32, PropModeReplace, 1, nxagentWindowPriv(pWin), 1) != Success)
+        fprintf(stderr, "nxagentReconnectWindow: Updating NX_REAL_WINDOW failed.\n");
+#ifdef DEBUG
```

Comment:
There is only ChangeWindowPropery(), even for creating a property that does not exist before. No such function as AddWindowProperty() exist in dix/.